bittensor/_subtensor/subtensor_impl.py

```diff
@@ -729,6 +762,197 @@ def add_stake(
+
+                if neuron.is_null:
+                    neurons.append( None )
+                    continue
+
+                neurons.append( neuron )
+
+        # Remove existential balance to keep key alive.
+        ## Keys must maintain a balance of at least 1000 rao to stay alive.
+        total_staking_rao = sum([amount.rao if amount is not None else 0 for amount in amounts])
+        if total_staking_rao == 0:
+            # Staking all to the first wallet.
+            if old_balance.rao > 1000:
+                old_balance -= bittensor.Balance.from_rao(1000)
+
+        elif total_staking_rao < 1000:
+            # Staking less than 1000 rao to the wallets.
+            pass
+        else:
+            # Staking more than 1000 rao to the wallets.
+            ## Reduce the amount to stake to each wallet to keep the balance above 1000 rao.
+            percent_reduction = 1 - (1000 / total_staking_rao)
+            amounts = [amount * percent_reduction for amount in amounts]
+        
+        successful_stakes = 0
+        for wallet, amount, neuron in zip(wallets, amounts, neurons):
+            if neuron is None:
+                bittensor.__console__.print(":cross_mark: [red]Hotkey: {} is not registered. Skipping ...[/red]".format( wallet.hotkey_str ))
+                continue
+
+            if wallet.coldkeypub.ss58_address != wallet_0.coldkeypub.ss58_address:
+                bittensor.__console__.print(":cross_mark: [red]Hotkey: {} is not under the same coldkey. Skipping ...[/red]".format( wallet.hotkey_str ))
+                continue
+
+            # Assign decrypted coldkey from wallet_0
+            #  so we don't have to decrypt again
+            wallet._coldkey = wallet_0._coldkey
+            staking_all = False
+            # Convert to bittensor.Balance
+            if amount == None:
+                # Stake it all.
+                staking_balance = bittensor.Balance.from_tao( old_balance.tao )
+                staking_all = True
+
+            elif not isinstance(amount, bittensor.Balance ):
+                staking_balance = bittensor.Balance.from_tao( amount )
+            else:
+                staking_balance = amount
+
+            # Estimate staking fee.
+            stake_fee = None # To be filled.
+            with bittensor.__console__.status(":satellite: Estimating Staking Fees..."):
+                with self.substrate as substrate:
+                    call = substrate.compose_call(
+                    call_module='SubtensorModule', 
+                    call_function='add_stake',
+                    call_params={
+                        'hotkey': wallet.hotkey.ss58_address,
+                        'ammount_staked': staking_balance.rao
+                        }
+                    )
+                    payment_info = substrate.get_payment_info(call = call, keypair = wallet.coldkey)
+                    if payment_info:
```

Comment:
Like this check! Do we have this check for the add_stake as well?

Reply:
yes, it's already there